File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/presentproof/common/Restriction.java
package com.evernym.verity.sdk.protocols.presentproof.common;

import com.evernym.verity.sdk.utils.AsJsonObject;
import org.json.JSONObject;

/**
 * A holder for a restriction expression
 */
public class Restriction implements AsJsonObject {
    final JSONObject data;

    protected Restriction(JSONObject data) {
        this.data = data;
    }

    /**
     * Convert this object to a JSON object
     *
     * @return this object as a JSON object
     */
    @Override
    public JSONObject toJson() {
        return this.data;
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/PresentProof/Common/Restriction.cs
using System.Json;
using VeritySDK.Utils;

namespace VeritySDK.Protocols.PresentProof
{
    /// <summary>
    /// A holder for a restriction expression
    /// </summary>
    public class Restriction : AsJsonObject
    {
        JsonObject data;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="data">Restrinction as SSON object</param>
        public Restriction(JsonObject data)
        {
            this.data = data;
        }

        /// <summary>
        /// Convert this object to a JSON object 
        /// </summary>
        /// <returns>this object as a JSON object</returns>
        public JsonObject toJson()
        {
            return this.data;
        }
    }
}